framework/test/unit/node/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -82,22 +86,217 @@ describe('CommitPool', () => {
+			when(bftParamsMock)
+				.calledWith(context, height + 1)
+				.mockResolvedValue(true);
+			// Act
+			await commitPool['_job'](context);
+			// Assert
+			// nonGossiped commits are moved to gossiped commits
+			expect([...commitPool['_nonGossipedCommits'].values()].flat()).toHaveLength(0);
+			expect([...commitPool['_gossipedCommits'].values()].flat()).toHaveLength(10);
+			expect(commitPool['_nonGossipedCommits'].get(1070)).toBeUndefined();
+		});
+
+		it('should clean all the commits from gossipedCommit that does not have bftParams change', async () => {
+			commitPool['_gossipedCommits'].set(1070, [
+				{
+					blockID: getRandomBytes(32),
+					certificateSignature: getRandomBytes(96),
+					height: 1070,
+					validatorAddress: getRandomBytes(20),
+				},
+			]);
+			// Assert
+			expect([...commitPool['_gossipedCommits'].values()].flat()).toHaveLength(7);
+			// Arrange
+			const bftParamsMock = jest.fn();
+			commitPool['_bftAPI'].existBFTParameters = bftParamsMock;
+			const context = createNewAPIContext(new InMemoryKVStore());
+			when(bftParamsMock).calledWith(context, 1071).mockResolvedValue(false);
+			when(bftParamsMock).calledWith(context, maxHeightCertified).mockResolvedValue(true);
+			when(bftParamsMock)
+				.calledWith(context, height + 1)
+				.mockResolvedValue(true);
+			// Act
+			await commitPool['_job'](context);
+			// Assert
+			// nonGossiped commits are moved to gossiped commits
+			expect([...commitPool['_nonGossipedCommits'].values()].flat()).toHaveLength(0);
+			expect([...commitPool['_gossipedCommits'].values()].flat()).toHaveLength(10);
+			expect(commitPool['_gossipedCommits'].get(1070)).toBeUndefined();
+		});
+
+		it('should select non gossiped commits that are created by the generator of the node', async () => {
+			const generatorAddress = getRandomBytes(20);
+			commitPool['_gossipedCommits'].set(1070, [
+				{
+					blockID: getRandomBytes(32),
+					certificateSignature: getRandomBytes(96),
+					height: 1070,
+					validatorAddress: generatorAddress,
+				},
+			]);
+			// Assert
+			expect([...commitPool['_gossipedCommits'].values()].flat()).toHaveLength(7);
+			// Arrange
+			commitPool['_bftAPI'].existBFTParameters = jest.fn().mockResolvedValue(true);
+			(commitPool['_generatorAddress'] as any) = generatorAddress;
+			const context = createNewAPIContext(new InMemoryKVStore());
+			// Act
+			await commitPool['_job'](context);
+			// Assert
+			// nonGossiped commits are moved to gossiped commits
+			expect([...commitPool['_nonGossipedCommits'].values()].flat()).toHaveLength(0);
+			expect([...commitPool['_gossipedCommits'].values()].flat()).toHaveLength(11);
+			expect(commitPool['_nonGossipedCommits'].get(1070)).toBeUndefined();
+			const commits = commitPool['_gossipedCommits'].get(1070);
+			expect(commits).toBeDefined();
+			expect(commits).toBeArray();
+			const generatorCommit = commits?.find(c => c.validatorAddress.equals(generatorAddress));
```

Comment:
This commit by generator is under `_gossipedCommits` but as per the description it says `select non gossiped commits`.
Maybe improve the description.

Reply:
good catch 😉